### DISCUSSION TITLE
Sonoff RF Bridge and Garage Door, Am I wasting my time ?

### DISCUSSION THREAD

**Discussion by BlueWings172** [arendst/Tasmota#14726] [Q&A] [answered]
Hello

I'm very new to Tasmota and home automation I've spent 2 days trying and researching the web.

I have a Sonoff RF Bridge that I have successfully flashed with both Tasmota and Portsich and all what I'm trying to achieve is to open/close my garage door using the Sonoff RF Bridge but I was unable to achieve this. What I was able to do (as a test) is to learn the code of a normal Sonoff remote (not my garage remote) and program one of the 16 Bridge button on Tasmota to replicate the code.

I ran the rfraw 177 command from Tasmota command and when I press the garage remote button, I can see a long code which I entered in the [BitBucket Converter](https://bbconv.hrbl.pl/) which did convert the B1 code to B0. When I run the rfraw command followed by the B0 code, nothing happens at all. I have read somewhere online that garage remotes have rolling codes which change all the time. 

So here is what I need to figure out:

1- Is this even possible or am I wasting my time ? My remote is an ATA SecureCode PTX4 (photo below).

2- What am I doing wrong ? Is there another step for remotes with rolling codes (don't know if mine has rolling code or not).  Here is an example of the code that the RF Bridge captured with the 177 command but it keeps changing a bit.

`{"RfRaw":{"Data":"AA B1 04 01B8 0F8C 033E 3E94 3808080808080808080808081A082A082A0A082A0A0A0828282828282A0828282A08282A082A082A0A082A0828282A0A0828282A0A082A082828282828282828282A08282828282A0A0A082A0A08 55"}}`

This what I sent from the console:

`rfraw AA B0 58 04 08 01B8 0F8C 033E 3E94 3808080808080808080808081A082A082A0A082A0A0A0828282828282A0828282A08282A082A082A0A082A0828282A0A0828282A0A082A082828282828282828282A08282828282A0A0A082A0A08 55`

This is what I see and the door does not open:

```
23:41:21.257 RSL: RESULT = {"RfRaw":"ON"}
23:41:22.348 RSL: RESULT = {"Time":"2022-02-04T23:41:22","RfRaw":{"Data":"AAA055"}}
```

I would really appreciate any help I could get. (please keep in mind I'm a complete beginner) 

Thanks

![20220203_115425](https://user-images.githubusercontent.com/65966061/152612685-571bc400-10b1-4168-b595-dd4d5eb16184.jpg)   ![20220203_115153](https://user-images.githubusercontent.com/65966061/152613303-5fb1a665-eb12-4e5c-b968-a33cbbc4842b.jpg)

![20220203_115135](https://user-images.githubusercontent.com/65966061/152613689-7d95168e-eb79-4d95-a5bd-202b6944fc09.jpg)

**Comment by sfromis** [comment, marked as answer]
Short answer: Yes, you *should* be wasting your time trying to do that.

The garage door opener would have a horribly bad security if you could make it work. Since I see "security" mentioned, the very simple stuff you can do without hacking tools (which the Sonoff RF Bridge is not, by far), should not work. It likely indeed has something like rolling codes which will not work with a simple "replay attack". If you really want to ignore security, you could get a garage door opener without security features, meaning that you should see the garage as "open access".

**Comment by rapi3** [comment]
You are wasting your time with RF bridge for any RF system that use "secure algorithm to send random RF codes".
You need a Tasmota device that control 1 or 2 relays installed in parallel with your actual system or relays will trigger open / close buttons of your remote control ( this is how I use it at this moment ).
It is more secure and you can access it as you planed to access the rfBridge by wifi/vpn.

**Reply by QuanchyPlimp** [reply]
I know this question is a few years old now. However, I managed to get my roller shutter to operate (433mhz) using RFBridge with Portisch. The roller shutter control box is around 25 years old, so probably not very secure.

It's worth sniffing as many codes as you can by pressing the button on your original remote, I did around 30 and decoded them with bitbucket. I had to try many times until I found the one that worked.

Also, with regard to the codes, you should add a ';' after 55 and add 'RfRaw 0' ~ this tells the bridge to leave sniffing mode so it can resume normal service; example, also add Backlog to the start of the code. The codes we generate/use cannot be stored to RFBridge memory so we use backlog;
 
Backlog RfRaw AA B0 58 04 08 01B8 0F8C 033E 3E94 3808080808080808080808081A082A082A0A082A0A0A0828282828282A0828282A08282A082A082A0A082A0828282A0A0828282A0A082A082828282828282828282A08282828282A0A0A082A0A08 55; RfRaw 0

Hope this helps someone.

**Comment by BlueWings172** [comment]
I was able to get it to work for 1 time. The garage system was turned off and I turned it on and tried the code and it did work once. Tried again a million times and it did not. The system is at least 15 years old so I don't think it is that secure and I don't need it to be. I also wanna avoid getting into replacing it as It works fine and I can't justify the cost.

I don't understand how it worked once and then stopped.

Have you tried Tasmota/Portsich with the Sonoff RF Bridge ?

**Reply by sfromis** [reply]
Rolling codes is basically confirmed by your one-time "success". The thing is that such systems needs to be able to handle lost codes, therefore they tend to have a small window of a few codes working at the same time. And a very basic security system like rolling codes matches the outdated RF 433 technology anyway. 

The Sonoff RF Bridge cannot be used for hacking of even this low security level. You are free to continue wasting your time :-D

**Reply by stefanbode** [reply]
TASMOTA "shutter & blinds" driver (setoption 80) has a well working garage mode to open and close the garage and have full TASMOTA control for it. You can then open/close it directly from your mobile without security issues.

**Reply by sfromis** [reply]
Of course, that very nice shutters/blinds feature will not work with the installed RF garage door opener with rolling codes. You may be able to wire a relay module to the control wires after the box taking care of security.

**Comment by BlueWings172** [comment]
Thank you all for your help ! Your explanation make perfect sense and I'm convinced that the system uses rolling codes and the RF Bridge is not the right tool to achieve my purpose.

maybe i should get one of these software defined radios dongles and try to figure out how to decode the code.

Just wondering if the RF Bridge is incapable of just decoding rolling code or also incapable of sending it (provided it was decoded by other means)? 

Also has anyone been successful with rolling codes with devices other than RF bridge?

**Reply by sfromis** [reply]
Googling for something like `sdr 433 rolling` does provide hits about people hacking such. Price goes up if want a SDR device able to send, not just receive. The RF bridge firmware on the RF chip does not care about what code it sends, as long as it is the right protocol (modulation), and within the bounds of what it can handle of size and timing.

It could easily be cheaper and faster if you work out how to connect a relay device to do the same controls as the current unit doing the security provides.

**Reply by barbudor** [reply]
The easiest and lazy way is to hook an ESP with 2 dry relays (or even transistors) to the buttons of the remote ....
Or if you have wall switches inside the house, a ESP device with a dry relay hooked on the buttons would do the same.

**Comment by BlueWings172** [comment]
The relay suggestion has been mentioned few times. I misunderstood that I need to connect AC relays to the motor directly and bypassing the RF system. Going back to those posts it is evident that it's not at all what is being suggested. It is easy to get lost when you are a beginner! I was also blinded by the false impression that decoding the remote and retransmitting the signal is gonna be a quick and easy thing. Foolish me !

I do have an underutilized ESP32 but I need to familiarize myself with dry relays and how to connect them to the remote.

Hopefully I will find some video or guide that would help.

Thanks all for the great idea.

**Reply by timaseth** [reply]
@BlueWings172 I have a Aliexpress Rolling gate opener with RF remote.
It was an easy Tasmota RF hack and I was concerned with the lack of security so I de-soldered the RF chip and just wired a 2 channel relay to a D1 mini and connected it to the push-button options on the main relay board - it all works a charm. Let us know if you get stuck along the way!
You will need wifi to be available near the door, that goes without saying.

**Reply by jksmurf** [reply]
Hi I appreciate this is an old post but is there any way you could explain what you did here in plain language please. I’d love to get this working thanks 🙏

**Comment by BlueWings172** [comment]
@jksmurf 

Have you found a solution that works for you?
If not, I can guide you. it's pretty simple once you have simple and cheap parts.

**Reply by jksmurf** [reply]
Hi @BlueWings172 thanks for coming back on this I’m probably going to go for a Shelly Wi-Fi switch connected to the door opener switch on the control unit, paired with some sensors at the doors.

cheers

k.
